cdap-common/src/main/java/io/cdap/cdap/common/namespace/NamespaceAdmin.java

```diff
@@ -63,4 +65,22 @@ public interface NamespaceAdmin extends NamespaceQueryAdmin {
    * @throws NamespaceNotFoundException if the specified namespace is not found
    */
   void updateProperties(NamespaceId namespaceId, NamespaceMeta namespaceMeta) throws Exception;
+
+  /**
+   * Computes unique namespace identity name by lowercase namespace id truncated to 15 characters
+   * appended with the first 15 hex characters of the namespace's SHA256.
+   *
+   * @param namespaceId the {@link NamespaceId} of the namespace.
+   * @return namespace unique identity name.
+   */
+  default String getIdentity(NamespaceId namespaceId) {
+    if (NamespaceId.isReserved(namespaceId.getNamespace())) {
+      return namespaceId.getNamespace();
```

Comment:
nit: Please add a comment saying this method shouldn't be applicable to the `system` and `cdap` namespaces, since they're also reserved namespaces.

Reply:
To remove confusion, changed the if statement to only check for `default` namespace.